rust/sealevel/client/src/router.rs

```diff
@@ -555,6 +566,82 @@ fn configure_owner(
     }
 }
 
+/// Idempotent. Attempts to set the upgrade authority to the intended owner if
+/// the payer can change the upgrade authority.
+fn configure_upgrade_authority(
+    ctx: &mut Context,
+    program_id: &Pubkey,
+    router_config: &RouterConfig,
+    chain_config: &ChainMetadata,
+) {
+    let client = chain_config.client();
+
+    let program_account = client.get_account(program_id).unwrap();
+    // If the program isn't upgradeable, exit
+    if program_account.owner == bpf_loader_upgradeable::id() {
```

Comment:
not too sure about solana specific stuff, but I would think equality would mean the program is upgradable

Reply:
this was a bug, great call